File: 1679-max-number-of-k-sum-pairs/1679-max-number-of-k-sum-pairs.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var maxOperations = function(nums, k) {
    nums.sort((a, b) => a - b); // Sort nums array in ascending order
    let count = 0;
    let left = 0;
    let right = nums.length - 1; 

    while (left < right) {
        let sum = nums[left] + nums[right];
        if (sum === k) {
            count++;
            left++;
            right--;
        } else if (sum < k) {
            left++;
        } else {
            right--;
        }
    }
    return count;
};

File: 1679-max-number-of-k-sum-pairs/1679-max-number-of-k-sum-pairs.rb
# @param {Integer[]} nums
# @param {Integer} k
# @return {Integer}
def max_operations(nums, k)
  count = 0
  numcount = Hash.new(0)
  nums.each do |num|
    complement = k - num
    if numcount[complement] > 0
        count+=1
        numcount[complement]-=1
    else
        numcount[num] +=1
    end
  end
  count
end